packages/gatsby/src/bin/cli.js

```diff
@@ -16,6 +17,18 @@ process.on(`unhandledRejection`, error => {
 
 const defaultHost = `localhost`
 
+let inGatsbySite = false
+let localPackageJSON
+try {
+  localPackageJSON = require(path.resolve(`./package.json`))
+  if (localPackageJSON.dependencies && localPackageJSON.dependencies.gatsby) {
```

Comment:
None :-) I just didn't think about that as I've never made gatsby a devDependency but can now there's plenty of use cases where that makes sense. I'll merge https://github.com/gatsbyjs/gatsby/pull/1738 and make a new release.